### DISCUSSION TITLE
FreeTube v0.22.0 Release Candidate

### DISCUSSION THREAD

**Discussion by efb4f5ff-1298-471a-8973-3d47447115dc** [FreeTubeApp/FreeTube#5881] [Announcements]
Hey everyone!

Today, we are "releasing" a test build of the upcoming release. These builds are a near final revision of what we plan to release for v0.22.0 and are close if not already stable. A Release Candidate build means that we are ready to start the final testing process and are close to a full release.

So that's where everyone else comes in. We have officially started our RC builds for v0.22.0 and they are now ready for testing. If anyone in the community would like to help, then they're more than welcome to do so. This thread will be for discussing these new builds as well as discussing any issues that may occur with them.

# Notes on Testing

- When you install a RC build, **this will overwrite your current version of FreeTube**. Please backup your subscriptions and other data before installing. While you will likely not have any data loss, it is best to do this anyways just in case.
- When testing, please focus on the changes mentioned in our patch notes. These are the main points that we are looking at for feedback. If there's a very important issue that needs addressed before the next release that isn't mentioned, then feel free to ask about it. If your most wanted feature was not included for this release, then please do not use this thread as a method of asking for it to be included.
- Please use this thread for reporting issues if you have found something wrong with the RC builds. **Do not create an issue in our issue board**.

# How To Install

Installation for RC builds are similar to nightly builds. If you've downloaded a nightly build before, then this will be no different. **You will need to be signed into GitHub in order to access these builds**. We would love to disable this however we are unable to.

- Go to our GitHub Actions tab. [This](https://github.com/FreeTubeApp/FreeTube/actions/workflows/build.yml?query=branch%3Av0.22.0-RC) link conveniently filters our workflows to only show builds for this RC.
- Click on the first build in the list that shows a green check mark next to it.
- Scroll down on the new page to see a list of available builds. Click on the one that matches your operating system of choice.
- This next step will vary on the OS that you're using. Install the downloaded file the same way you normally would for a full release (**Again, this will overwrite your current stable installation of FreeTube**)

At this point, when you open up FreeTube you will now be using the RC build. You can confirm this by checking out the About tab and making sure that "RC" is included in the version number that's displayed.

Changes / Patch Notes will be listed below.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
# Current changelog since v0.21.3
> [!NOTE]
> Styling / formatting are subject to change between now and the full release.

## Summary

Our Video Player has undergone a massive overhaul. We have switched from video.js to shaka-player. This allows us to add allot of new features and fix many issues. Notably, it's now possible to watch higher quality video streams. A massive thank you to @absidue for their help and support on this!

This is a very large change to how we handle things. If you happen to see anything that seems broken compared to the last release, please consider commenting about it in the thread below so that we can look at it.

For users hoping to import their subscriptions from YouTube, the process should be much quicker and a lot smoother since we no longer fetch thumbnail information when importing from YouTube. The channel thumbnails in the sidebar will be populated as you navigate around the app.

We would also like to thank all the translators that make FreeTube more accessible to all of our users!

## Features
* Migrate video player from video.js to shaka-player by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/4978
* Local API: Use IOS HLS manifest for livestreams by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5705
* Add a videos tab to artist topic channels by using YouTube's auto-generated playlists on the local API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5661
* Implement persistent subscription cache by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5185
* Add ability to see comment replies of community posts through Invidious by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5589
* Display video tags in search results with the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5590
* Add support for Invidious instances that require the Basic authentication scheme to access them by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5569
* Implement case sensitive search in history by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5446
* Proxy live streams when "Proxy Videos Through Invidious" is enabled by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5649
* Add support for copying hashtag and post links from the context menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5538 and https://github.com/FreeTubeApp/FreeTube/pull/5760
* Add a GPU Internals entry to the View menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5550
* Add support for Belarusian locale by @aliaksandr-trush in https://github.com/FreeTubeApp/FreeTube/pull/5841
* Add Playlist Sort By Video Duration by @Hoverth in https://github.com/FreeTubeApp/FreeTube/pull/5627

## Fixes
* Don't make requests when importing subscriptions by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5617
* Update playlist import to only add duplicate playlist items if the playlist already contained duplicates by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5783
* Fix the shorts tab being empty on the Local API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5679
* Fix persistent cache incorrectly having higher priority then auto fetch feed option by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5743
* Fix video start time handling by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5719
* Fix long comments overflowing and breaking the layout by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5774
* Local API: Fix error when a community post has no likes by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5759
* Fix loading from search cache failing when no features are selected by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5775
* Fix subscriptions CSV import shortening channel names by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5793
* Fix hashtag page not using Invidious API when it should by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5794
* fix `cmd+m` shortcut to minimize the window on macos and not mute the video by @jacopo-degattis in https://github.com/FreeTubeApp/FreeTube/pull/5829
* Fix saving Invidious thumbnail URLs for subscriptions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5662
* Fix about page not reacting to locale changes by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5393
* Do not overwrite cache entries when hitting RSS ratelimits by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5756
* Fix playback of VOD/non-live videos with Invidious by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5654
* Fix audio track selection by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5697
* Properly handle Arabic hashtags by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5545
* Fix a few memory leaks while tearing down the player by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5698
* Lowercase hls manifest type to match the parser name shaka player registers by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/5648
* Fix wrong live stream playlist used for subscriptions by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5876

## Changes
* Bump electron from 31.1.0 to 32.2.0
* Change the Proxy Videos Through Invidious setting to only apply when using the Invidious API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5758
* Remove "Force Local Backend for Legacy Formats" setting by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5650
* Revert "Larger search/URL bar" by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5773
* Settings menu & UX improvements by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5029
* Allow users to pass their own hardware acceleration flags on the command line on Linux by @xsmile in https://github.com/FreeTubeApp/FreeTube/pull/5438
* Update icon for channels button on side nav by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5273
* Display a better error message on the watch page when YouTube has blocked the IP address by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5757
* Change Sponsorblock Default Category Colors by @Neuheit in https://github.com/FreeTubeApp/FreeTube/pull/5686
* Update RSS tooltip by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5753
* Clarify how to add subscriptions by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5778
* Update subscription cache when subscribing from the channel page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5717
* Update top nav to move query text to new window instead of remaining in old window by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5541
* Fix incorrect/translated placeholders in locale files by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5628
* Update subscription details from search results by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5795
*  Various performance improvements by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5522, https://github.com/FreeTubeApp/FreeTube/pull/5666 and https://github.com/FreeTubeApp/FreeTube/pull/5722
* Various code cleanups by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5520, https://github.com/FreeTubeApp/FreeTube/pull/5776, https://github.com/FreeTubeApp/FreeTube/pull/5663 and https://github.com/FreeTubeApp/FreeTube/pull/5625

## Known Issues
When using the Local API or the Invidious API it is possible that you will run into issues due to YouTube IP blocks. Please try to use a different Invidious Instance, VPN or proxy when you encounter those issues.

## New Contributors
* @xsmile made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5438
* @Neuheit made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5686
* @jacopo-degattis made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5829
* @aliaksandr-trush made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5841
* @Hoverth made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5627

**Full Changelog**: https://github.com/FreeTubeApp/FreeTube/compare/v0.21.3-beta...v0.22.0-RC

**Comment by jn64** [comment]
Minor feedback on the new player:

1. Resolution and playback speed are no longer visible without mouseover or opening the respective menus.
2. Auto resolution no longer tells you the actual resolution in use.
3. New controls are a wide margin away from the edge of the video. This is less convenient, especially in full screen. Previously I could flick my mouse to the bottom edge/corners.

Old:
![ft-old](https://github.com/user-attachments/assets/97f475b4-2d57-4706-bb98-951a4b52dee2)
New:
![ft-new](https://github.com/user-attachments/assets/6eb8c953-d17b-433a-b3f2-b4e6a6356dc7)

**Comment by ACR-Jeff** [comment]
In some videos comments are disabled and not visible in this version "There are no comments available for this video"
While in previous release the same videos will have the comments enabled and visible.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
This issue was introduced right after the RC, https://github.com/FreeTubeApp/FreeTube/issues/5880. It been fixed and will be included in the upcoming release.

**Comment by jn64** [comment]
Seeking forward near the end of the video never reaches the end. If the remaining time is less than the seek interval, it moves to 1s before the end.

With `L` or right arrow seeking by time, this is slightly odd behaviour but not really a problem. However the old player and YT web/mobile all seek to the end.

With `.` seeking by frame, at the last frame, it will jump back 1s.

v0.22.0-RC-4923 Beta rpm on Fedora 39

**Reply by absidue** [reply]
Does it happen on all videos or only some videos? If it only happens on certain videos, could you please provide links to them?

**Reply by jn64** [reply]
Every video that I've tried (avc, av1, and vp9). Tested different lengths from 10s to 1hr+ videos. I use local API only.

Screencast showing the problem. The video here is https://youtu.be/jNQXAC9IVRw

[rec-2024-10-22_00.33.46.webm](https://github.com/user-attachments/assets/b4309c57-ddb4-4eb3-9aa1-ea4e6a19038b)

1. I let the video play until the end and stop.
2. I press right arrow once; it jumps to 1s from the end. I press right arrow a second time; it remains at 1s from the end.
3. I play the video again and press right arrow three times to show it jumping back.
4. With the video stopped at the last frame, I press `.` and it jumps to 1s from the end.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Thanks that video was very insightful, I can reproduce this on my end

**Comment by jn64** [comment]
Main window sometimes has an unnecessary horizontal scrollbar due to elements temporarily exceeding the bounds.

Repro:
1. Enable theatre mode
2. Fullscreen a video (press `f`)
3. Mouseover the progress bar near the end of the video, to show a thumbnail on the right side
4. Exit fullscreen (press `f` again). While exiting fullscreen, you can see the thumbnail briefly appear far off to the right.
5. The window now has a horizontal scrollbar.

The bug is not specific to theatre mode, but depending on the window size, the thumbnail might not appear far enough to cause horizontal scrolling in non-theatre mode.

v0.22.0-RC-4923 Beta rpm on Fedora 39

Screencast (ignore the blank thumbnails, that is a YouTube problem for this video):

[rec-2024-10-22_17.54.23-freetube-horizontal-scrollbar.webm](https://github.com/user-attachments/assets/1d3e07c6-3b06-42ab-abb8-5bd33dc54273)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Can reproduce!

**Reply by absidue** [reply]
Opened https://github.com/FreeTubeApp/FreeTube/pull/5929 to fix it.

**Comment by octimis** [comment]
Would it be possible to please remove the pause/play icon in the middle of the video? It makes tutorial videos less feasible.
![frtube](https://github.com/user-attachments/assets/30089507-f82e-4cac-a52c-e36bb2069608)

**Reply by ChunkyProgrammer** [reply]
I believe going to `Settings > Player Settings > Display Play Button In Video Player` and setting it to off will fix your issue
